packages/docusaurus/src/client/exports/Link.tsx

```diff
@@ -142,7 +142,7 @@ function Link({
       {...props}
       onMouseEnter={onMouseEnter}
       innerRef={handleRef}
-      to={targetLink}
+      to={targetLink || ''}
```

Comment:
`to` has always been a `string | undefined` and if it goes through the `undefined` path then it ends up a `<a` and not down this branch, as the linter doesn't allow a `!` then I opted for the 'should never occur' `|| ""`